nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-standard-record-utils/src/main/java/org/apache/nifi/schema/access/SchemaAccessStrategy.java

```diff
@@ -38,4 +40,11 @@
      * @return the set of all Schema Fields that are supplied by the RecordSchema that is returned from {@link #getSchema(FlowFile, InputStream)}.
      */
     Set<SchemaField> getSuppliedSchemaFields();
+
+    /**
+     * @return Whether this factory needs an incoming FlowFile to resolve Record Schema.
+     */
+    default boolean isFlowFileRequired() {
```

Comment:
@ijokarumawak I very much agree with the idea behind this JIRA. However, I think I would approach the solution in a somewhat different manner. Rather than having a boolean indicate whether or not a FlowFile is required, I would recommend we get rid of the FlowFile being passed in to get the schema and instead just pass in a Map<String, String>. This way, if you have no FlowFile you can just pass in an empty Map. It also allows for more flexibility so that you can pass in a Map<String, String> where the schema.name may be set by the processor, for instance, instead of an empty Map.